OasisDEX/oasis-borrow
# Not enough funds for tx

# [Not enough funds for tx](https://app.shortcut.com/oazo-apps/story/4382/hardware-wallet-error-handling-not-enough-funds-to-cover-gas-fees)

<please insert a shortcut link above>
  
## Changes 👷‍♀️
  <Please add short list of changes>
- added warning and error saying that user doesn't have enough funds to pay for transaction
  
## How to test 🧪
  <Please explain how to test your changes>

- testing using ledger, trezor -> verify both (warning and error) by trying to open borrow / multiply / earn vault. You can use wallet with 0 ETH and by hardcoding `depositAmountExceedsCollateralBalance` constant to `false` in conditions you will be able to go to proxy step and try to send tx.

- testing using metamask or other software wallets - test it using for example hardhat / goerli, push your wallet balance to the point where estimated gas (USD) will be higher than your ETH balance (USD)
